admin/client.py

```diff
@@ -235,14 +243,76 @@ def postOptions(self):
             build_server=self['build-server'],
         )
 
-        if self['pip']:
-            supported = PIP_DISTRIBUTIONS
-        else:
-            supported = PACKAGED_CLIENT_DISTRIBUTIONS
-        if self['distribution'] not in supported:
-            raise UsageError(
-                "Distribution %r not supported. Available distributions: %s"
-                % (self['distribution'], ', '.join(supported)))
+
+def get_steps_pip(distribution, package_source=PackageSource()):
+    """
+    Get commands to run for testing client pip installation.
+
+    :param bytes distribution: The distribution the node is running.
+    :param PackageSource package_source: The source from which to install the
+        package.
+
+    :return: An ``Effect`` to pass to a ``Dispatcher`` that supports
```

Comment:
Not for now but this interface seems to be needed enough that perhaps it should be an explicit interface